mlflow/store/sqlalchemy_store.py

```diff
@@ -0,0 +1,222 @@
+            experiment_id=experiment_id, is_deleted=False).first()
+
+        if not exp:
+            raise MlflowException('No Experiment with id={} exists'.format(experiment_id),
+                                  error_codes.RESOURCE_DOES_NOT_EXIST)
+
+        return exp.to_mlflow_entity()
+
+    def delete_experiment(self, experiment_id):
+        exp = self.session.query(models.SqlExperiment).get(experiment_id)
+        exp.is_deleted = True
+        self._save_to_db(exp)
+
+    def restore_experiment(self, experiment_id):
+        exp = self.session.query(models.SqlExperiment).get(experiment_id)
+        exp.is_deleted = False
+        self._save_to_db(exp)
+
+    def rename_experiment(self, experiment_id, new_name):
+        experiment = self.session.query(models.SqlExperiment).get(experiment_id)
+        experiment.name = new_name
+        self._save_to_db(experiment)
+
+    def create_run(self, experiment_id, user_id, run_name, source_type, source_name,
+                   entry_point_name, start_time, source_version, tags, parent_run_id):
+        _ = parent_run_id
+        experiment = self.get_experiment(experiment_id)
+
+        if experiment.lifecycle_stage != entities.Experiment.ACTIVE_LIFECYCLE:
+            raise MlflowException('Experiment id={} must be active'.format(experiment_id),
+                                  error_codes.INVALID_STATE)
+
+        run = models.SqlRun(name=run_name, artifact_uri=None,
+                            experiment_id=experiment_id, source_type=source_type,
+                            source_name=source_name, entry_point_name=entry_point_name,
+                            user_id=user_id, status=entities.RunStatus.RUNNING,
+                            start_time=start_time, end_time=None,
+                            source_version=source_version,
+                            lifecycle_stage=entities.RunInfo.ACTIVE_LIFECYCLE)
+
+        for tag in tags:
+            run.tags.append(models.SqlRunTag(key=tag.key, value=tag.value))
+        self._save_to_db([run])
+
+        run = run.to_mlflow_entity()
+
+        return run
+
+    def update_run_info(self, run_uuid, run_status, end_time):
+        run = self.session.query(models.SqlRun).filter_by(run_uuid=run_uuid).first()
+        run.status = run_status
+        run.end_time = end_time
+
+        self._save_to_db(run)
+        run = run.to_mlflow_entity()
+
+        return run.info
+
+    def restore_run(self, run_id):
+        run = self.session.query(models.SqlRun).filter_by(run_uuid=run_id).first()
+        run.is_deleted = False
+        self._save_to_db(run)
+
+    def get_run(self, run_uuid):
+        run = self.session.query(models.SqlRun).filter_by(run_uuid=run_uuid,
+                                                          is_deleted=False).first()
+        if run is None:
+            raise MlflowException('Run(uuid={}) doesn\'t exist'.format(run_uuid),
+                                  error_codes.RESOURCE_DOES_NOT_EXIST)
+
```

Comment:
get params, tags, and metrics (for each metric key - latest timestamped metric value)